File: WebApi/Models/File.cs

public class File : IModel
{
    public int Id { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    public string FileKeyString { get; set; } = string.Empty;
    public int? HorseId { get; set; }
    public Horse? Horse { get; set; }

}

File: vue-app/src/classes/File.ts
import { LocalIDFactory } from "@/helpers/LocalIDFactory";
import { IFile } from "@/interfaces/IFile";

export class File implements IFile {
  localID?: string;
  id?: number;
  name: string;
  createdAt: Date;
  updatedAt: Date;
  fileKeyString: string;
  fileKeyStrings: string[];
  horseId?: number;

  constructor() {
    this.localID = LocalIDFactory.createLocalID();
    this.name = "Datei";
    this.createdAt = new Date();
    this.updatedAt = new Date();
    this.fileKeyString = "";
    this.fileKeyStrings = [];
  }
}
